### ISSUE TITLE
List all public repositories

### ISSUE THREAD

**Issue by shiftkey** [octokit/octokit.net#686]
Reference: https://developer.github.com/v3/repos/#list-all-public-repositories 

- [x] `GetAllPublic()`
- [x]  `GetAllPublic(int since)`

**Comment by rms81** [comment]
Hi,

Do you mind that I try to implement this one?

**Comment by shiftkey** [comment]
![](http://i.imgur.com/fwRaskw.jpg)

**Comment by rms81** [comment]
Pull request created.

**Comment by alfhenrik** [comment]
Does the `GetAllPublic(int since)` still need implementing, or is there still something to figure out in regards to Pagination?

**Comment by shiftkey** [comment]
@alfhenrik the endpoint which takes a `since` doesn't exist, so I've left this open.

Pagination is also different for the public repository APIs:

```
Link: <https://api.github.com/repositories?since=364>; rel="next"
```

Instead of the default:

```
Link: <https://api.github.com/resource?page=2>; rel="next",
      <https://api.github.com/resource?page=5>; rel="last"
```

But we can deal with that in #760 or :boot: to a subsequent PR.

**Comment by shiftkey** [comment]
Reopening this one as the work done in #774 neglected to catch a scenario with pagination that was specific to this API:

 - the `Link` has a `since` parameter older than the specified id
 - when invoking the next link, `GetAll` will overwrite the parameter with the original id
 - forever paging :sob:

If you revert 26a8bf0e818ad8a620987e13d5ed7b1b8bacfa01 you'll get this PR back at least.

**Comment by shiftkey** [comment]
Shipped in 0.11.0
